### ISSUE TITLE
CookieMiddleware cookies with existing name are added instead of updated

### ISSUE THREAD

**Issue by Tetragramat** [kriswallsmith/Buzz#392]
This results in Cookie header, together with issue #391, to look like this: 
`Cookie: PHPSESSID=37f0tkl68vu8k4bgboqdep3ti2, PHPSESSID=7tmrmb9uebe92rgf6t0fju9a25`

**Comment by kids-return** [comment]
I also have the same problem

https://github.com/kriswallsmith/Buzz/blob/62422cbbd17efa757353116d42508fda8503c754/lib/Middleware/Cookie/CookieJar.php#L61

```php
    /**
     * Processes Set-Cookie headers from a request/response pair.
     */
    public function processSetCookieHeaders(RequestInterface $request, ResponseInterface $response): void
    {
        $host = $request->getUri()->getHost();
        foreach ($response->getHeader('Set-Cookie') as $header) {
            $cookie = new Cookie();
            $cookie->fromSetCookieHeader($header, $host);

            if ($this->existsName($cookie)) {
                $this->setCookie($cookie);
            } else {
                $this->addCookie($cookie);
            }
        }
    }
```
This may be right，Need to add the `existsName` and `setCookie` methods

**Comment by kids-return** [comment]
You can generate a hash based on `$cookie`.
E.g.
```php
    /**
     * Adds a cookie to the current cookie jar.
     */
    public function addCookie(Cookie $cookie): void
    {
        $hash = sha1($cookie->getName() . $cookie->getAttribute('domain') . $cookie->getAttribute('path') . $cookie->getAttribute('http'));
        $this->cookies[$hash] = $cookie;
    }
```
This makes it easy to implement `exists` and `setCookie`.
If this solution works, you can give me feedback here, I will fix this problem and submit `PR`

**Comment by Tetragramat** [comment]
I droped use of CookieMiddleware so I'm unable to test it, but it looks like it should work.
Only make sure co use correct attribute names https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#Directives
I see no directive with name `http`.

**Comment by Tetragramat** [comment]
I can not find any official source for updating cookies, but for example Guzzle tests name, path and domain so it should be ok with only these.
https://github.com/guzzle/guzzle/blob/master/src/Cookie/CookieJar.php#L182

**Comment by kids-return** [comment]
@Nyholm @kriswallsmith 
Can I fix it like this?

**Comment by Nyholm** [comment]
Yes, Modifying addCookie makes sense. I'll be happy to see a PR. 

Could you also include your fix from #391?

**Comment by kids-return** [comment]
Yes, I will improve the relevant PR as soon as possible
